proxy/http/HttpTransactCache.cc

```diff
@@ -949,17 +949,17 @@ HttpTransactCache::calculate_quality_of_accept_encoding_match(MIMEField *accept_
   // still okay, but otherwise, this is just not a match at all.         //
   /////////////////////////////////////////////////////////////////////////
   if ((q == -1.0) && is_identity_encoding) {
-    if (match_gzip(accept_field) == GZIP) {
-      if (match_gzip(cached_accept_field) == GZIP) {
-        return (float)1.0;
+    if (match_content_encoding(accept_field, "gzip")) {
+      if (match_content_encoding(cached_accept_field, "gzip")) {
+        return float(1);
```

Comment:
Also, why float(1) ? Why not either of

```
static_cast<float)(1)
```

or, perhaps (would have to test)

```
1f
```

? Reading float(1), it makes me think it's a function, but I don't think it is.

Reply:
It's another way of doing a cast:  http://en.cppreference.com/w/cpp/language/explicit_cast

I think in C++PL, Stroustrup said he considers C-style casts to be deprecated.  He says this type of cast is better because it the same syntax as conversion using the constructor of a user-defined type.